src/types/createRouteOptions.ts

```diff
@@ -0,0 +1,126 @@
+import { Component } from 'vue'
+import { combineName } from '@/services/combineName'
+import { combinePath, CombinePath } from '@/services/combinePath'
+import { combineQuery, CombineQuery } from '@/services/combineQuery'
+import { AfterRouteHook, BeforeRouteHook } from '@/types/hooks'
+import { Host } from '@/types/host'
+import { ExtractParamTypes, MergeParams } from '@/types/params'
+import { Path, ToPath } from '@/types/path'
+import { Query, ToQuery } from '@/types/query'
+import { RouteMeta } from '@/types/register'
+import { Route } from '@/types/route'
+import { MaybeArray } from '@/types/utilities'
+
+/**
+ * Defines route hooks that can be applied before entering, updating, or leaving a route, as well as after these events.
+ */
+type WithHooks = {
+  onBeforeRouteEnter?: MaybeArray<BeforeRouteHook>,
+  onBeforeRouteUpdate?: MaybeArray<BeforeRouteHook>,
+  onBeforeRouteLeave?: MaybeArray<BeforeRouteHook>,
+  onAfterRouteEnter?: MaybeArray<AfterRouteHook>,
+  onAfterRouteUpdate?: MaybeArray<AfterRouteHook>,
+  onAfterRouteLeave?: MaybeArray<AfterRouteHook>,
+}
+
+type WithComponent = {
+  /**
+   * A Vue component, which can be either synchronous or asynchronous components.
+   */
+  component: Component,
+}
+
+type WithComponents = {
+  /**
+   * Multiple components for named views, which can be either synchronous or asynchronous components.
+   */
+  components: Record<string, Component>,
+}
+
+type WithComponentCallback<Params = unknown> = {
+  loadComponent: (params: Params) => any,
+}
```

Comment:
Lets chat about this. I'm not quite following the implementation you'd do to use this. But I see the params types 🎉

Reply:
We removed this because we plan on changing it anyway.